bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tmva/_gnn.py

```diff
@@ -0,0 +1,175 @@
+            for j in shape_as_list:
+                shape.push_back(j)
+            gin.globals_update_block.GetFunctionBlock().AddInitializedTensor['float'](i.name, gbl_namespace.TMVA.Experimental.SOFIE.ETensorType.FLOAT, shape, i.numpy())
+    else:
+        print("Invalid Model for global update.")
+        return
+
+    # adding edge-node aggregate function
+    edge_node_reducer = GraphModule._node_block._received_edges_aggregator._reducer.__qualname__
+    if(edge_node_reducer == "unsorted_segment_sum"):
+        agg = gbl_namespace.TMVA.Experimental.SOFIE.RFunction_Sum()
+        gin.createAggregateFunction[gbl_namespace.TMVA.Experimental.SOFIE.RFunction_Sum](agg, gbl_namespace.TMVA.Experimental.SOFIE.FunctionRelation.NODES_EDGES)
+    elif(edge_node_reducer == "unsorted_segment_mean"):
+        agg = gbl_namespace.TMVA.Experimental.SOFIE.RFunction_Mean()
+        gin.createAggregateFunction[gbl_namespace.TMVA.Experimental.SOFIE.RFunction_Mean](agg, gbl_namespace.TMVA.Experimental.SOFIE.FunctionRelation.NODES_EDGES)
+    else:
+        print("Invalid aggregate function for edge-node reduction")
+        return
+
+
+    # adding node-global aggregate function
+    node_global_reducer = GraphModule._global_block._nodes_aggregator._reducer.__qualname__
+    if(node_global_reducer == "unsorted_segment_sum"):
+        agg = gbl_namespace.TMVA.Experimental.SOFIE.RFunction_Sum()
+        gin.createAggregateFunction[gbl_namespace.TMVA.Experimental.SOFIE.RFunction_Sum](agg, gbl_namespace.TMVA.Experimental.SOFIE.FunctionRelation.NODES_GLOBALS)
+    elif(node_global_reducer == "unsorted_segment_mean"):
+        agg = gbl_namespace.TMVA.Experimental.SOFIE.RFunction_Mean()
+        gin.createAggregateFunction[gbl_namespace.TMVA.Experimental.SOFIE.RFunction_Mean](agg, gbl_namespace.TMVA.Experimental.SOFIE.FunctionRelation.NODES_GLOBALS)
+    else:
+        print("Invalid aggregate function for node-global reduction")
+        return
+
+    # adding edge-global aggregate function
+    node_global_reducer = GraphModule._global_block._edges_aggregator._reducer.__qualname__
+    if(node_global_reducer == "unsorted_segment_sum"):
+        agg = gbl_namespace.TMVA.Experimental.SOFIE.RFunction_Sum()
+        gin.createAggregateFunction[gbl_namespace.TMVA.Experimental.SOFIE.RFunction_Sum](agg, gbl_namespace.TMVA.Experimental.SOFIE.FunctionRelation.EDGES_GLOBALS)
+    elif(node_global_reducer == "unsorted_segment_mean"):
+        agg = gbl_namespace.TMVA.Experimental.SOFIE.RFunction_Mean()
+        gin.createAggregateFunction[gbl_namespace.TMVA.Experimental.SOFIE.RFunction_Mean](agg, gbl_namespace.TMVA.Experimental.SOFIE.FunctionRelation.EDGES_GLOBALS)
+    else:
+        print("Invalid aggregate function for node-global reduction")
+        return
+
+
+    gnn_model = gbl_namespace.TMVA.Experimental.SOFIE.RModel_GNN(gin)
+    blas_routines = gbl_namespace.std.vector['std::string']()
+    blas_routines.push_back("Gemm")
```

Comment:
For future reference,
Here `Gemm` libraries are added by default from the Parsers, since the MLP update functions are only implemented and used now. Later, we should have this dependent on the requirements of the update and aggregate functions.